build.gradle.kts

```diff
@@ -422,9 +424,86 @@ subprojects {
         }
     }
 
+    plugins.withId("me.champeau.gradle.japicmp") {
+        afterEvaluate {
+            // hack to find the current released version of the project
+            val temp: Configuration = project.configurations.create("tempConfig")
+            project.dependencies.add("tempConfig", "io.opentelemetry:opentelemetry-bom:latest.release")
+            val moduleVersion = project.configurations["tempConfig"].resolvedConfiguration.firstLevelModuleDependencies.elementAt(0).moduleVersion
+            project.configurations.remove(temp)
+
+            val userRequestedBase = project.properties["apiBaseVersion"]
+            val baselineVersion = userRequestedBase ?: moduleVersion
+            val newVersion : String? = project.properties["apiNewVersion"] as String?
+            val baselineArtifact: File?
+            var newArtifact: File? = null
+            val oldGroup = project.group
+            try {
+                //this dance is needed due to gradle not allowing more than one version of a dependency.
+                project.group = "virtual_baseline_for_japicmp"
+                val depModule = "io.opentelemetry:${base.archivesBaseName}:$baselineVersion@jar"
+                val depJar = "${base.archivesBaseName}-${baselineVersion}.jar"
+                val configuration: Configuration = configurations.detachedConfiguration(
+                        dependencies.create(depModule)
+                )
+                baselineArtifact = files(configuration.files).filter {
+                    it.name.equals(depJar)
+                }.singleFile
+            } finally {
+                project.group = oldGroup
+            }
+            if (newVersion != null) {
+                try {
+                    project.group = "virtual_new_for_japicmp"
+                    val depModule = "io.opentelemetry:${base.archivesBaseName}:$newVersion@jar"
+                    val depJar = "${base.archivesBaseName}-${newVersion}.jar"
+                    val configuration: Configuration = configurations.detachedConfiguration(
+                            dependencies.create(depModule)
+                    )
+                    newArtifact = files(configuration.files).filter {
+                        it.name.equals(depJar)
+                    }.singleFile
+                } finally {
+                    project.group = oldGroup
+                }
+            }
+
+            tasks {
+                val jar = getByName("jar") as Jar
+                if (newArtifact == null) {
+                    newArtifact = file(jar.archiveFile)
+                }
+                create<JapicmpTask>("japicmp") {
```

Comment:
```suggestion
                val japicmp by registering<JapicmpTask> {
```

Reply:
where do I get `registering` from? IDEA doesn't seem to know it's a thing

Reply:
or, rather, IDEA sees two of these `registering` delegate functions, but neither of them apply

Reply:
aha. I figured it out. It takes the task class as a parameter.